### ISSUE TITLE
Bad auto-generated files for `hook_set_value`

### ISSUE THREAD

**Issue by kunaltyagi** [colcon/colcon-core#515]
I'm using colcon to build ROS1 stack from scratch, and observed the following logs under `set -vex` while performing a `source /opt/ros/noetic/setup.bash`:
```
...                                                                                                                     
+++ . /opt/ros/noetic/ros_environment/share/ros_environment/catkin_env_hook/1.ros_version.sh                                                                                                                       
# generated from ros_environment/env-hooks/1.ros_version.sh.in                                                                                                                                                     
                                                                                                                                                                                                                   
export ROS_VERSION=1                                                                                                                                                                                               
++++ export ROS_VERSION=1                                                                                                                                                                                          
++++ ROS_VERSION=1                                                                                                                                                                                                 
COLCON_CURRENT_PREFIX="/opt/ros/noetic/rosbash" _colcon_prefix_sh_source_script "/opt/ros/noetic/rosbash/share/rosbash/hook/catkin_env_hook_workspace.bash"                                                        
+++ COLCON_CURRENT_PREFIX=/opt/ros/noetic/rosbash                                                                                                                                                                  
+++ _colcon_prefix_sh_source_script /opt/ros/noetic/rosbash/share/rosbash/hook/catkin_env_hook_workspace.bash                                                                                                      
+++ '[' -f /opt/ros/noetic/rosbash/share/rosbash/hook/catkin_env_hook_workspace.bash ']'                                                                                                                           
+++ '[' -n '' ']'                                                                                                                                                                                                  
+++ . /opt/ros/noetic/rosbash/share/rosbash/hook/catkin_env_hook_workspace.bash                                                                                                                                    
# generated from colcon_bash/shell/template/hook_set_value.bash.em                                                                                                                                                 
                                                                                                                                                                                                                   
export CATKIN_ENV_HOOK_WORKSPACE="$COLCON_PREFIX_PATH"                                                                                                                                                             
++++ export CATKIN_ENV_HOOK_WORKSPACE=/opt/ros/noetic                                                                                                                                                              
++++ CATKIN_ENV_HOOK_WORKSPACE=/opt/ros/noetic                                                                                                                                                                     
COLCON_CURRENT_PREFIX="/opt/ros/noetic/rosbash" _colcon_prefix_sh_source_script "/opt/ros/noetic/rosbash/share/rosbash/catkin_env_hook/15.rosbash.bash"                                                            
+++ COLCON_CURRENT_PREFIX=/opt/ros/noetic/rosbash                                                                                                                                                                  
+++ _colcon_prefix_sh_source_script /opt/ros/noetic/rosbash/share/rosbash/catkin_env_hook/15.rosbash.bash                                                                                                          
+++ '[' -f /opt/ros/noetic/rosbash/share/rosbash/catkin_env_hook/15.rosbash.bash ']'                                                                                                                               
+++ '[' -n '' ']'
+++ . /opt/ros/noetic/rosbash/share/rosbash/catkin_env_hook/15.rosbash.bash
# generated from rosbash/env-hooks/15.rosbash.bash.em

if [ -z "$CATKIN_ENV_HOOK_WORKSPACE" ]; then
  CATKIN_ENV_HOOK_WORKSPACE="/opt/ros/noetic/rosbash"
fi
++++ '[' -z /opt/ros/noetic ']'
. "$CATKIN_ENV_HOOK_WORKSPACE/share/rosbash/rosbash" 
++++ . /opt/ros/noetic/share/rosbash/rosbash                                                                                                                                             
bash: /opt/ros/noetic/share/rosbash/rosbash: No such file or directory
```

Interesting things to note:
* `export CATKIN_ENV_HOOK_WORKSPACE="$COLCON_PREFIX_PATH"` seems to be bad. There is no package path. The correct value of `ENV_HOOK_WORKSPACE` is `/opt/ros/noetic/<pkg_name>`
* This issue could be fixed by use of `COLCON_CURRENT_PREFIX` instead (but I'm not sure)
* Issue is present in colcon-bash as well (same hook set template file)

**Comment by kunaltyagi** [comment]
This issue goes away with:
```sh
grep 'WORKSPACE="$COLCON_PREFIX_PATH"' ${ROS_INSTALL_DIR} -lr > bad_files.txt &&
cat bad_files.txt &&
sed -i 's/WORKSPACE="\$COLCON_PREFIX_PATH"/WORKSPACE="$COLCON_CURRENT_PREFIX"/' $(cat bad_files.txt)
```

**Comment by cottsay** [comment]
I believe this was fixed in #549, which will be part of the next release.
